### PR TITLE
[ISSUE #1840] Migrated to FlatList

### PR DESCRIPTION
**Pull request by jeluard**
addresses #1840

### Summary:

Migrated all list components (deprecated in recent RN releases) to new FlatList component.

### Steps to test:

- Sanity check status, especially all `lists`

Test opened chat very throughly, including order of messages, if the new messages are added at the correct place and whether :load-more event is still correctly firing once somebody scrolls at the very top of the message list (test by opening chat with more then 20 messages, scroll to the very top and check if all the messages are loaded and rendered in correct order)

status: ready

### CODE REVIEW

**Comment by jeluard** [comment]
Blocked by #1690

**Comment by oskarth** [comment]
@jeluard 
> TODO Chat list inverted behavior is broken as the package we used to rely on doesn't work with FlatList. Fortunately latest RN (starting 0.47) support this feature ootb.

Can we minimize the scope of this PR to partially migrate things? As opposed to leaving big PR open for long time. Upgrading to 0.47 is non-trivial and will take a while. Plus there might even be a circular dependency.

**Comment by jeluard** [comment]
@oskarth Makes sense I will do that.

**Comment by oskarth** [comment]
@jeluard What's the state of this PR?

**Comment by jeluard** [comment]
Waiting for RN upgrade which should be finished soon.

**Comment by janherich** [comment]
@jeluard please rebase on current develop

File: src/status_im/ui/components/react.cljs [hunk 1]
```diff
@@ -127,6 +121,8 @@
 (defn linear-gradient [props]
   [gradient props])
 
+
+;; TODO deprecate
```
Comment (flexsurfer) [inline src/status_im/ui/components/react.cljs]: deprecated?

File: src/status_im/ui/components/list/views.cljs [hunk 2]
```diff
@@ -18,97 +18,101 @@
...
+
+(def section-separator [react/view styles/section-separator])
+
+(defn- base-list-props
+  [{:keys [render-fn empty-component header separator default-separator?]}]
+  (let [separator (or separator (when (and platform/ios? default-separator?) default-separator))]
+    (merge {:keyExtractor (fn [_ i] i)}
+           (when render-fn               {:renderItem (wrap-render-fn render-fn)})
+           (when separator               {:ItemSeparatorComponent (fn [] (reagent/as-element [separator]))})
+           ;; TODO(jeluard) Does not work with our current ReactNative version
```
Comment (flexsurfer) [inline src/status_im/ui/components/list/views.cljs]: still doesn't work?

File: src/status_im/chat/screen.cljs [hunk 3]
```diff
@@ -76,35 +76,37 @@
...
-     [chat-toolbar]
-     [messages-view group-chat]
-     [input/container {:text-empty? (string/blank? input-text)}]
-     (when show-actions?
-       [actions/actions-view])
-     (when show-bottom-info?
-       [bottom-info/bottom-info-view])
-     [offline/offline-view {:top (get-in platform/platform-specific
-                                         [:component-styles :status-bar :default :height])}]]))
+           {:component-did-mount    #(re-frame/dispatch [:check-and-open-dapp!])
```
Comment (flexsurfer) [inline src/status_im/chat/screen.cljs]: indentation

**Review by flexsurfer — APPROVED** [review]

**Comment by janherich** [comment]
@jeluard Looks good, just please add following test description:
* Test opened chat very throughly, including order of messages, if the new messages are added at the correct place and whether `:load-more` event is still correctly firing once somebody scrolls at the very top of the message list (test by opening chat with more then 20 messages, scroll to the very top and check if all the messages are loaded and rendered in correct order)

**Review by janherich — APPROVED** [review]

File: src/status_im/ui/components/list/views.cljs (hunk 2 above)
Comment (jeluard) [reply src/status_im/ui/components/list/views.cljs]: It does!

**Review by dmitryn — APPROVED** [review]

**Comment by asemiankevich** [comment]
Branch: feature/#1840
Android: https://i.diawi.com/zhR9ZF
iOS: https://i.diawi.com/Z6cE2z

1. Weird message at the top in Console chat when creating account. I checked last nighly for this - looks good.

https://app.testfairy.com/projects/4803622-status/builds/7346240/sessions/1/?accessToken=gI-ufGQVb0ceElq1c2blsaTY9Sk

image

2. Blank screen when clicking `/password` command. Also does not happening in nightly

https://app.testfairy.com/projects/4803590-status/builds/7344703/sessions/6/?accessToken=8I0jsAlF/urL7WJuyJ7xI-2yKTs

image

3. Blank screen when tapping `enter your phone` message

image


https://app.testfairy.com/projects/4803622-status/builds/7346240/sessions/3/?accessToken=iyiUq-qqZgIgUxbzi0kU/Q3K8GU

**Comment by asemiankevich** [comment]
Branch: status-im:feature/#1840
Android: https://i.diawi.com/emj3kM
iOS: https://i.diawi.com/W1XoPL

The same empty window when clicking /password message upon launch


image


https://app.testfairy.com/projects/4803590-status/builds/7397032/sessions/16/?accessToken=oGy-6BGXqFITv2ryzdgm2E3EQu8

https://app.testfairy.com/projects/4803622-status/builds/7421993/sessions/1/?accessToken=EfjCwwaMmOG945ij9sHmsUec2mU

**Pull request merged**